### ISSUE TITLE
S3Client.getBucketAcl() issue

### ISSUE THREAD

**Issue by sahandilshan** [aws/aws-sdk-java-v2#1488]
Calling S3Client.putBucketAcl() with an AccessControlPolicy triggers a S3Exception with error message "The XML you provided was not well-formed or did not validate against our published schema".

## Expected Behavior
PutBucketACL requests should complete successfully.

## Current Behavior
The request fails with an S3Exception and message "The XML you provided was not well-formed or did not validate against our published schema"

## Steps to Reproduce (for bugs)
This is the code I used,
```
            // add new grantee to acl
            Grantee grantee = Grantee.builder()
                    .type(Type.CANONICAL_USER)
                    .id("xxxxxxxxxxxxxxxxxxxxxxxxxxxxx") // canonical id
                    .build();
            Grant newGrantee = Grant.builder()
                    .grantee(grantee)
                    .permission(Permission.WRITE)
                    .build();

            AccessControlPolicy acl = AccessControlPolicy.builder()
                    .grants(newGrantee)
                    .build();
            //put the new acl
            PutBucketAclRequest putAclReq = PutBucketAclRequest.builder()
                    .bucket(bucketName)
                    .accessControlPolicy(acl)
                    .build();
            s3.putBucketAcl(putAclReq);
```
error log `software.amazon.awssdk.services.s3.model.S3Exception: The XML you provided was not well-formed or did not validate against our published schema (Service: S3, Status Code: 400, Request ID: 757BCD0BC3C4E01E)`

## Your Environment
* AWS Java SDK version used: 2.10.1
* JDK version used: 1.8
* Operating System and version: Ubuntu 18.04.3 LTS

**Comment by sahandilshan** [comment]
I tried this one too, but end up with the same exception
``` 
List<Grant> grantList = new ArrayList<>(getAclRes.grants());
            String ownerCanonicalId = getAclRes.grants().get(0).grantee().id();
            Grant newGrantee = Grant.builder()
                    .grantee(grantee)
                    .permission(Permission.WRITE)
                    .build();
            grantList.add(newGrantee);

            AccessControlPolicy acl = AccessControlPolicy.builder()
                    .owner(builder -> builder.id(ownerCanonicalId))
                    .grants(grantList)
                    .build();
```

**Comment by dagnir** [comment]
Thank you for the issue report. We will be taking a look.

**Comment by sahandilshan** [comment]
```
       List<Grant> grantList2 = new ArrayList<>();
          Grant ownerGrant = Grant.builder()
                .grantee(builder -> {
                    builder.id(ownerCanonicalId)
                            .type(Type.CANONICAL_USER);
                })
                .permission(Permission.FULL_CONTROL)
                .build();
        grantList2.add(ownerGrant);
        grantList2.add(newGrantee);
        AccessControlPolicy acl = AccessControlPolicy.builder()
                .owner(builder -> builder.id(ownerCanonicalId))
                .grants(grantList2)
                .build();
        //put the new acl
        PutBucketAclRequest putAclReq = PutBucketAclRequest.builder()
                .bucket(bucketName)
                .accessControlPolicy(acl)
                .build();
        try {
            s3.putBucketAcl(putAclReq);
        } catch (S3Exception e) {
            logger.error("Error while adding ACL permission to the bucket ", e);
        }
```
I was able to fix the issue with this aproach, it seems like the issue is with the List that returns from 
`getAclRes.grants()` method

**Comment by zoewangg** [comment]
@sahandilshan Yeah, there is a known issue with `s3#getObjectAcl` - the `Grantee#type` is not unmarshalled correctly and the field is null. It might cause the error if the field is being used in the `putObjectAclRequest`

In your first repro code, did you also use the grants retrieved from the `GetBucketAclResponse`?

**Comment by sahandilshan** [comment]
```
GetBucketAclRequest bucketAclReq = GetBucketAclRequest.builder()
                    .bucket(bucketName)
                    .build();
GetBucketAclResponse getAclRes = s3.getBucketAcl(bucketAclReq);
```
@zoewangg Yes, I used `GetBucketAclResponse` to retriev grants

**Comment by dagnir** [comment]
This should be fixed by #1563. Resolving.
